### PR TITLE
(GH-1582) added constraint to require source value when adding source

### LINKED ISSUE #1582: Choco source add allows empty url
Steps to reproduce on version 0.10.11:
`choco source add -n="sourcename"`
results in source without url address
`sourcename - | Priority 0|Bypass Proxy - False|Self-Service - False|Admin Only - False`

Expected result:
Creation of source without an address should be prohibited, since it doesn't make sense.

### PR DESCRIPTION
**Pull request by jirkapok**
* Added new constrain when creating source
* Added notes to help messages
* Added related unit test

Closes #1582

### CODE REVIEW

**Comment by jirkapok** [comment]
Codacy issue to dont throw ApplicationException was added because i follow other usage in the same method.

File: src/chocolatey/infrastructure.app/commands/ChocolateyFeatureCommand.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public virtual void configure_argument_parser(OptionSet optionSet, ChocolateyCon
 
             optionSet
                 .Add("n=|name=",
-                     "Name - the name of the source. Required with some actions. Defaults to empty.",
+                     "Name - the name of the source. Required with actions other than list. Defaults to empty.",
```
Comment (ferventcoder) [inline src/chocolatey/infrastructure.app/commands/ChocolateyFeatureCommand.cs]: Wouldn't mind if this was split to a different commit

**Comment by ferventcoder** [comment]
One other nitpick - git commits:

* summary should be 50 characters 
* one empty line
* commit body where you explain changes breaking at 72 characters

This is in our contributing doc (yes, we are kind of picky)

**Comment by gep13** [comment]
@jirkapok thanks again for creating this PR.  I have taken the liberty of making the corrections that were suggested by @ferventcoder as well as rebasing your changes onto the head of the current master branch.

@ferventcoder this should be ready for a final review, prior to merging.

**Comment by jirkapok** [comment]
thanks @gep13 , i was lazy to do so.

**Comment by gep13** [comment]
This was merged into stable at https://github.com/chocolatey/choco/commit/548196a74de1d24b63e6f2bd6d959fbe77bb8cfd

**Comment by gep13** [comment]
Merged into stable to be released with 0.10.12.

**Comment by gep13** [comment]
@jirkapok your changes have been merged, thanks for your contribution 👍
